### ISSUE TITLE
Hardcoded 8GB limit in the FilterConsensusReads processes

### ISSUE THREAD

**Issue by lbeltrame** [nf-core/fastquorum#57]
### Description of the bug

There is a hardcoded limit of 8GB in the FilterConsensusReads processes defined in `modules.config`. Aside being on the low side for very deep sequenced datasets (think ctDNA), it relies on retries to increase memory requirements should the task fail. However, what happens with many schedulers like SLURM is that if the task goes out of memory (and with 10,000X data it is a likely possibility) it is killed, causing an error and the premature end of the pipeline (so it won't ever be retried).

I would think this should belong to the user's definitions and should not be hardcoded. 

### System information

- Nextflow: 23.04.2
- Hardware: HPC
- Executor: slrum
- Container Engine: apptainer
- OS: Debian
- nf-core/fastquorum version: 1.0.0

**Comment by lbeltrame** [comment]
No, I get exit status 137.  I assume because it went over the memory limit and it was killed. 

It does retry, but it ultimately errors out.

```
Caused by:                                                                                                                        
  Process `NFCORE_FASTQUORUM:FASTQUORUM:FASTQC (5571767_P4)` terminated with an error exit status (137)                           
                                                                                                                                  
                                                                                                                                  
Command executed:                                                                                                                 
                                                                                                                                  
  printf "%s %s\n" 5571767_P4_S1_R1_001.fastq.gz 5571767_P4_1.gz 5571767_P4_S1_R2_001.fastq.gz 5571767_P4_2.gz | while read old_na
me new_name; do                                                                                                                   
      [ -f "${new_name}" ] || ln -s $old_name $new_name                                                                           
  done                                                                                                                            
                                                                                                                                  
  fastqc \                                                                                                                        
      --quiet \                                                                                                                   
      --threads 8 \                                                                                                               
      --memory 8192 \                                                                                                             
      5571767_P4_1.gz 5571767_P4_2.gz                                                                                             
                                                                                                                                  
  cat <<-END_VERSIONS > versions.yml                                                                                              
  "NFCORE_FASTQUORUM:FASTQUORUM:FASTQC":                                                                                          
      fastqc: $( fastqc --version | sed '/FastQC v/!d; s/.*v//' )                                                                 
  END_VERSIONS                                                                                                                    
                                                                                                                                  
Command exit status:                                                                                                              
  137                                                                                                                             
                                                                                                                                  
Command output:                                                                                                                   
  application/gzip                                                                                                                
  application/gzip                                                                                                                
                                                                                                                                  
Command error:                                                                                                                    
  .command.sh: line 10:    47 Killed                  fastqc --quiet --threads 8 --memory 8192 5571767_P4_1.gz 5571767_P4_2.gz    
                                                                                                                                  
Work dir:                                                                                                                         
  /XXXX/work/36/731674114933ce9ebd1d8fb24fd827                                    
                                                                                                                                  
Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line           
                                                                                                                                  
 -- Check '.nextflow.log' file for details  
```

**Comment by lbeltrame** [comment]
Notice that there is a similar problem with FastQC. I had to make an override to raise the memory limit to 12G to make sure it finished.

**Comment by nh13** [comment]
I'd be glad to pointed where you think the config should live.   Is the user config you're talking about in `conf/base.config`?  Given the `FilterConsensusReads` process is a module-level config, why isn't it appropriate to have it in the [`conf/modules.config`](https://github.com/nf-core/fastquorum/blob/cbb0055e63c7f12fe25aacb58f1c9bd294c5af82/conf/modules.config#L111)?

Of course, users are always free to override any of the configs in this pipeline, which is by design.  I think it's reasonable that folks would need to customize the resources based on their input data and compute environment.  What am I missing>

**Comment by SPPearce** [comment]
Hi Nils,
I think the issue is that the memory is fixed at 8Gb, and doesn't increase when the module fails because it doesn't have the user retry part.

**Comment by SPPearce** [comment]
Fixed in #60
